src/hotspot/share/cds/aotClassLinker.cpp

```diff
@@ -143,7 +143,7 @@ bool AOTClassLinker::try_add_candidate(InstanceKlass* ik) {
   if (ik->is_hidden()) {
     assert(ik->shared_class_loader_type() != ClassLoader::OTHER, "must have been set");
     if (!CDSConfig::is_dumping_invokedynamic()) {
-      return false;
+      return true;
```

Comment:
I think this should be 
```
if (!CDSConfig::is_dumping_methodhandles()) {
  return false;
}
```

Because `is_dumping_methodhandles()` can generate hidden classes that are not lambda proxies.

Reply:
Fixed.